### ISSUE TITLE
Fatal Error

### ISSUE THREAD

**Issue by rgrinberg** [ocaml/ocaml-lsp#234]
Relevant stack trace:

```
[Error - 5:11:18 PM] Request textDocument/documentSymbol failed.
  Message: File "ocaml-lsp-server/src/position.ml", line 15, characters 4-10: Assertion failed
  Code: -32603 
/-----------------------------------------------------------------------
| Internal error: Uncaught exception.
| File "ocaml-lsp-server/src/position.ml", line 15, characters 4-10: Assertion failed
| Raised at file "vendor/stdune/result.ml" (inlined), line 19, characters 15-22
| Called from file "ocaml-lsp-server/src/document.ml", line 84, characters 2-19
| Called from file "vendor/fiber/fiber.ml", line 187, characters 36-41
| Called from file "vendor/fiber/fiber.ml", line 98, characters 8-15
\-----------------------------------------------------------------------
```

@ulugbekna do you want to take a stab at this?

**Comment by ulugbekna** [comment]
Sure

**Comment by johnhaley81** [comment]
This bug is killing me! Anything I can do to help out?

**Comment by ulugbekna** [comment]
I'm on the busier side lately. If you have time, go ahead and find the reason why this is happening or how to reproduce. 

Sorry for the delay!

**Comment by rgrinberg** [comment]
Np. I pushed a quick fix in #242. I don't know the source of these bad positions, but at least we don't have to crash.
